src/django_scim/middleware.py

```diff
@@ -49,7 +49,7 @@ def process_request(self, request):
         # If we've just passed through the auth middleware and there is no user
         # associated with the request we can assume permission
         # was denied and return a 401.
-        if not hasattr(request, 'user') or not get_is_authenticated_predicate()(request.user):
+        if not get_is_authenticated_predicate()(request):
             if request.path.startswith(self.reverse_url):
```

Comment:
These tests should happen in the other order, to reduce unnecessary extra work for all the other endpoints in the project.

For that matter, though, why does SCIM even need to have a middleware? I would think it’d be enough to just check authentication in a decorator used on all the SCIM endpoints. Then we wouldn’t need to slow down the rest of the project at all.

Reply:
> For that matter, though, why does SCIM even need to have a middleware?

This project was built under specific constraints and having a middleware fit those constraints. PR's are welcome if you'd like to speed things up.

Reply:
> These tests should happen in the other order, to reduce unnecessary extra work for all the other endpoints in the project.

Not sure I follow here. For the default predicate, `not hasattr(request, 'user') or request.user.is_annoymous` is faster than the `request.path.startswith`. But this is Python and web framework so we are talking microseconds compared to multiple millseconds of a request. I think clarity of code makes more sense in these situations. IMO, the failing soon for non-SCIM specific reasons makes more sense to me.

Reply:
> This project was built under specific constraints and having a middleware fit those constraints. PR's are welcome if you'd like to speed things up.

Okay, here’s one proposal: we could move `SCIMAuthCheckMiddleware` out of the global project settings and apply it directly to `SCIMView.dispatch`. This fixes the performance issue on non-SCIM views with minimal restructuring.

- #85

Reply:
> Not sure I follow here. For the default predicate, `not hasattr(request, 'user') or request.user.is_annoymous` is faster than the `request.path.startswith`.

No, it’s isn’t. `request.user` is a [lazy object](https://github.com/django/django/blob/4.0.6/django/contrib/auth/middleware.py#L25). Forcing its evaluation will typically involve a round trip to a database server or at least a cache server. On some endpoints that would have happened anyway, but on others it would not.